hyperlink/_url.py

```diff
@@ -1464,3 +1504,329 @@ def remove(self, name):
         """
         return self.replace(query=((k, v) for (k, v) in self.query
                                    if k != name))
+
+
+EncodedURL = URL  # An alias better describing what the URL really is
+
+
+class DecodedURL(object):
+    """DecodedURL is a type meant to act as a higher-level interface to
+    the URL. It is the `unicode` to URL's `bytes`. `DecodedURL` has
+    almost exactly the same API as `URL`, but everything going in and
+    out is in its maximally decoded state. All percent decoding is
+    handled automatically.
+
+    Where applicable, a UTF-8 encoding is presumed. Be advised that
+    some interactions, can raise UnicodeEncodeErrors and
+    UnicodeDecodeErrors, just like when working with
+    bytestrings.
+
+    Examples of such interactions include handling query strings
+    encoding binary data, and paths containing segments with special
+    characters encoded with codecs other than UTF-8.
+    """
+    def __init__(self, url, lazy=False):
```

Comment:
If this is a public class, then its initializer should be documented.  Please include an explanation of `url` and `lazy` in the docstring.